File: classic_server.py
import btfpy

def callback(node,data,len):
  print("Received: " + data.decode())
  if data[0] == ord("H"):
    btfpy.Write_node(node,"Hello world\n",0)
    print("Disconnecting...")
    return(btfpy.SERVER_EXIT)
  else:
    btfpy.Write_node(node,"Send Hello to exit\n",0)
    return(btfpy.SERVER_CONTINUE)
  
if btfpy.Init_blue("devices.txt") == 0:
  exit(0)

print("IF FAILS - Experiment with security = 0/1/2/3")
security = 3

keyflag = btfpy.KEY_ON | btfpy.PASSKEY_LOCAL
if security == 1:
  keyflag = btfpy.KEY_OFF | btfpy.PASSKEY_LOCAL
elif security == 2:
  keyflag = btfpy.KEY_OFF | btfpy.PASSKEY_OFF
elif security == 3:
  keyflag = btfpy.KEY_ON | btfpy.PASSKEY_OFF    

btfpy.Classic_server(btfpy.ANY_DEVICE,callback,10,keyflag)
btfpy.Close_all()
  

File: classic_server.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "btlib.h"

int callback(int node,unsigned char *data,int len);

int main()
  {
  int security,keyflag;
  
  if(init_blue("devices.txt") == 0)
    return(0);
  
  printf("IF FAILS - Experiment with security = 0/1/2/3\n");
  security = 3;

  keyflag = KEY_ON | PASSKEY_LOCAL;
  if(security == 1)
    keyflag = KEY_OFF | PASSKEY_LOCAL;  
  else if(security == 2)
    keyflag = KEY_OFF | PASSKEY_OFF;
  else if(security == 3)
    keyflag = KEY_ON | PASSKEY_OFF;    

  classic_server(ANY_DEVICE,callback,10,keyflag);
  close_all();
  }
  
  
int callback(int node,unsigned char *data,int len)
  {
  static unsigned char *xmessage = {"Hello world\n"};
  static unsigned char *message = {"Send Hello to exit\n"};
  
  printf("Received: %s",data);
  if(data[0] == 'H')
    {
    write_node(node,xmessage,strlen(xmessage));
    printf("Disconnecting...\n");
    return(SERVER_EXIT);
    }
  write_node(node,message,strlen(message));
  return(SERVER_CONTINUE);
  }
